### DISCUSSION TITLE
MOVED error

### DISCUSSION THREAD

**Discussion by lavrd** [redis/rueidis#320] [Q&A]
Hi, I have Redis cluster in AWS, but I sometimes I have following error: `MOVED 10254 ...-0001-001....:6379`. I found something similar in https://github.com/redis/rueidis/issues/55, but you spoke in this issue about dedicated connections, I don't use them. Can you suggest a solution, please?

**Comment by rueian** [comment]
Hi @lavrd, that is weird. The `MOVED` error should be handled [here](https://github.com/redis/rueidis/blob/20875c372a849cd3cc31585758afa8e5b2fa5850/cluster.go#L810). Could you give me more information? Such as:
* How did you setup Redis cluster on AWS
* Did you use `DoMulti` or just `Do`
* What Redis command you used

**Reply by lavrd** [reply]
Hi, sorry for such late reply.

- We have ElasticCache cluster with two nodes
- In this case I was using Do
- Command:
'''
s.rclient.B().Set().Key("").Value("").Px(time.Hour * 24 * 7).Build()
'''

The problem started after change some setting in AWS ElasticCache.

After application restart the problem gone.

Please write me if you need more info.

**Reply by rueian** [reply]
Hi @lavrd,

Could you share the response of `CLUSTER SLOTS` commands from your cluster?

**Reply by rueian** [reply]
I just tried failover on ElasticCaches with and without cluster mode, but unfortunately, I couldn't reproduce your case. In both modes, the MOVED errors were handled automatically.

**Reply by lavrd** [reply]
Unfortunately we are not using cluster anymore, but looks like it was issues on our side. Thanks for your time.
